src/SignalRServiceExtension/TriggerBindings/SignalRFilterAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+#pragma warning disable CS0618 // Type or member is obsolete
+    public abstract class SignalRFilterAttribute : FunctionInvocationFilterAttribute
+    {
+        public override Task OnExecutingAsync(FunctionExecutingContext executingContext,
+            CancellationToken cancellationToken)
+        {
+            if (executingContext.Arguments.FirstOrDefault().Value is InvocationContext invocationContext)
+            {
+                return FilterAsync(invocationContext, cancellationToken);
+            }
+            // Should not hit the Exception.
+            throw new InvalidOperationException($"{nameof(FunctionExceptionContext)} doesn't contain {nameof(InvocationContext)}.");
+        }
+
+        /// <summary>
+        /// Executed before the Function method being executed.
+        /// Throwing exceptions can terminate the Function execution and response the invocation failure.
+        /// </summary>
+        public abstract Task FilterAsync(InvocationContext invocationContext, CancellationToken cancellationToken);
```

Comment:
Can we make the filter returns true/false?

Reply:
The message of exception will be sent back to client side. So, I think let customer to throw exception maybe better. Or if they returns false, what should we throw?

Reply:
I remember the exception throwed in function will respond 500 internal server error to client, so we can throw in OnExecutingAsync?

Reply:
No, it the behavior of httpTrigger. In SignalRTrigger, we handle the exception and send back the invocation response with message.